jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -497,7 +497,16 @@ ecma_op_create_dynamic_function (const ecma_value_t *arguments_list_p, /**< argu
   {
     fallback_proto = ECMA_BUILTIN_ID_GENERATOR;
 
-    if (parse_opts & ECMA_PARSE_ASYNC_FUNCTION)
+    if (parse_opts & ECMA_PARSE_ASYNC_FUNCTION && !(parse_opts & ECMA_PARSE_GENERATOR_FUNCTION))
+    {
+      fallback_proto = ECMA_BUILTIN_ID_ASYNC_FUNCTION_PROTOTYPE;
+
+      if (new_target_p == NULL)
+      {
+        new_target_p = ecma_builtin_get (ECMA_BUILTIN_ID_ASYNC_FUNCTION);
+      }
+    }
+    else if (parse_opts & ECMA_PARSE_ASYNC_FUNCTION)
```

Comment:
I would put the above block inside the if, so no need to check `(parse_opts & ECMA_PARSE_ASYNC_FUNCTION)` twice